# Project 6 -- Ritvik Indupuri

**TA Help:** John Smith, Alice Jones

- Help with figuring out how to write a function.
    
**Collaboration:** Friend1, Friend2
    
- Helped figuring out how to load the dataset.
- Helped debug error with my plot.

## Question 1

In [3]:
%sql sqlite:////anvil/projects/tdm/data/lahman/lahman.db

In [4]:
%%sql
-- Checking the Salaries table
SELECT * FROM Salaries LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1985, 'ATL', 'NL', 'barkele01', 870000.0),
 (1985, 'ATL', 'NL', 'bedrost01', 550000.0),
 (1985, 'ATL', 'NL', 'benedbr01', 545000.0),
 (1985, 'ATL', 'NL', 'campri01', 633333.0),
 (1985, 'ATL', 'NL', 'ceronri01', 625000.0)]

In [5]:
%%sql
-- Checking the Teams table
SELECT * FROM Teams LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1871, 'NA', 'BS1', 'BNA', '', 3, 31, '', 20, 10, '', '', 'N', '', 401, 1372, 426, 70, 37, 3, 60, 19, 73, 16, '', '', 303, 109, 3.55, 22, 1, 3, 828, 367, 2, 42, 23, 243, 24, 0.834, 'Boston Red Stockings', 'South End Grounds I', '', 103, 98, 'BOS', 'BS1', 'BS1'),
 (1871, 'NA', 'CH1', 'CNA', '', 2, 28, '', 19, 9, '', '', 'N', '', 302, 1196, 323, 52, 21, 10, 60, 22, 69, 21, '', '', 241, 77, 2.76, 25, 0, 1, 753, 308, 6, 28, 22, 229, 16, 0.829, 'Chicago White Stockings', 'Union Base-Ball Grounds', '', 104, 102, 'CHI', 'CH1', 'CH1'),
 (1871, 'NA', 'CL1', 'CFC', '', 8, 29, '', 10, 19, '', '', 'N', '', 249, 1186, 328, 35, 40, 7, 26, 25, 18, 8, '', '', 341, 116, 4.11, 23, 0, 0, 762, 346, 13, 53, 34, 234, 15, 0.818, 'Cleveland Forest Citys', 'National Association Grounds', '', 96, 100, 'CLE', 'CL1', 'CL1'),
 (1871, 'NA', 'FW1', 'KEK', '', 7, 19, '', 7, 12, '', '', 'N', '', 137, 746, 178, 19, 8, 2, 33, 9, 16, 4, '', '', 243, 97, 5.17, 19, 1, 0, 507, 261, 5, 21, 17, 163, 8, 0.803, 'Fort Wayne Kekiongas', 'Hamilton Field', '', 101, 107, 'KEK', 'FW1', 'FW1'),
 (1871, 'NA', 'NY2', 'NNA', '', 5, 33, '', 16, 17, '', '', 'N', '', 302, 1404, 403, 43, 21, 1, 33, 15, 46, 15, '', '', 313, 121, 3.72, 32, 1, 0, 879, 373, 7, 42, 22, 235, 14, 0.84, 'New York Mutuals', 'Union Grounds (Brooklyn)', '', 90, 88, 'NYU', 'NY2', 'NY2')]

In [6]:
%%sql
-- Checking the Batting table
SELECT * FROM Batting LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('aardsda01', 2004, 1, 'SFN', 'NL', 11, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2006, 1, 'CHN', 'NL', 45, '', 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0),
 ('aardsda01', 2007, 1, 'CHA', 'AL', 25, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
 ('aardsda01', 2008, 1, 'BOS', 'AL', 47, '', 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0),
 ('aardsda01', 2009, 1, 'SEA', 'AL', 73, '', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)]

1..2 Name any columns that appear in multiple tables or only in a specific table
Shared Columns: playerID, yearID, and teamID are common keys that appear in multiple tables, linking them together.

Unique Columns: The Salaries table has a unique salary column. The Teams table has columns like Rank and attendance. The Batting table has many unique statistic columns like H (Hits), HR (Home Runs), and RBI.

1.3 Describe how the tables might relate to each other
The tables are linked through their shared ID columns. This is a relational database structure. For example, I can use playerID to connect a player's salary from the Salaries table to their performance stats in the Batting table for a given yearID.



## Question 2

In [7]:
%%sql
-- Joining Salaries and People on their common playerID
SELECT * FROM Salaries
JOIN People ON Salaries.playerID = People.playerID
LIMIT 5;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1985, 'ATL', 'NL', 'barkele01', 870000.0, 'barkele01', 1955, 7, 7, 'Fort Knox', 'USA', 'KY', '', '', '', '', '', '', 'Len', 'Barker', 'Leonard Harold', 225, 77, 'R', 'R', '1976-09-14', 'barkele01', '1987-09-26', 'barkl001'),
 (1985, 'ATL', 'NL', 'bedrost01', 550000.0, 'bedrost01', 1957, 12, 6, 'Methuen', 'USA', 'MA', '', '', '', '', '', '', 'Steve', 'Bedrosian', 'Stephen Wayne', 200, 75, 'R', 'R', '1981-08-14', 'bedrost01', '1995-08-09', 'bedrs001'),
 (1985, 'ATL', 'NL', 'benedbr01', 545000.0, 'benedbr01', 1955, 8, 18, 'Birmingham', 'USA', 'AL', '', '', '', '', '', '', 'Bruce', 'Benedict', 'Bruce Edwin', 175, 73, 'R', 'R', '1978-08-18', 'benedbr01', '1989-09-11', 'beneb001'),
 (1985, 'ATL', 'NL', 'campri01', 633333.0, 'campri01', 1953, 6, 10, 'Trion', 'USA', 'GA', 2013, 4, 25, 'USA', 'GA', 'Rydal', 'Rick', 'Camp', 'Rick Lamar', 195, 73, 'R', 'R', '1976-09-15', 'campri01', '1985-10-05', 'campr001'),
 (1985, 'ATL', 'NL', 'ceronri01', 625000.0, 'ceronri01', 1954, 5, 19, 'Newark', 'USA', 'NJ', '', '', '', '', '', '', 'Rick', 'Cerone', 'Richard Aldo', 192, 71, 'R', 'R', '1975-08-17', 'ceronri01', '1992-07-10', 'ceror001')]

In [8]:
%%sql
-- Creating a temporary table 'joined_data' with the combined info
WITH joined_data AS (
    SELECT * FROM Salaries JOIN People ON People.playerID = Salaries.playerID
)
-- Querying the temporary table to get the average salary by weight
SELECT
    weight,
    AVG(salary) AS avg_salary
FROM joined_data
GROUP BY weight
ORDER BY avg_salary DESC
LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(300, 14576190.4),
 (267, 13457901.642857144),
 (275, 10164293.103448275),
 (249, 8855550.0),
 (268, 6714593.75),
 (228, 6090648.0),
 (285, 5617744.680851064),
 (239, 4493312.5),
 (270, 4128199.1228070175),
 (242, 4105041.6666666665)]

The JOIN command is fundamental for leveraging the power of a relational database. My first query successfully merged player biographical data (like names from People) with their financial data (salary from Salaries). The second query demonstrates a more practical application: by joining the tables, I was able to analyze the relationship between a physical attribute (weight) and compensation. The results, showing the average salary for different weight groups, could be the first step in analyzing if there's a correlation between a player's physique and their earnings. Using a WITH clause made the final query much more readable and manageable.



## Question 3

In [9]:
%%sql
-- Joining People and Batting to link names to stats
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)
-- Summing all runs ('R') for the player named Babe Ruth
SELECT SUM(R)
FROM joined_data
WHERE nameFirst = 'Babe' AND nameLast = 'Ruth';

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(2174,)]

In [10]:
%%sql
-- Joining People and Batting
WITH joined_data AS (
    SELECT * FROM People JOIN Batting ON People.playerID = Batting.playerID
)
-- Filtering for players born in the 1980s
-- Then calculating the average runs, grouped and ordered by their birth year
SELECT
    birthYear,
    AVG(R) as avg_runs
FROM joined_data
WHERE birthYear >= 1980 AND birthYear <= 1989
GROUP BY birthYear
ORDER BY birthYear;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[(1980, 17.10182767624021),
 (1981, 14.335890878090366),
 (1982, 16.083526682134572),
 (1983, 16.491117478510027),
 (1984, 13.758029978586723),
 (1985, 12.153616703952274),
 (1986, 13.602470930232558),
 (1987, 14.006301197227474),
 (1988, 9.934713375796179),
 (1989, 12.811985018726592)]

These queries show how joining tables allows for deep, specific analysis. The first query successfully calculated Babe Ruth's career runs (2174), demonstrating how to aggregate data for a single entity across many years of records. The second query provided a demographic snapshot, revealing trends in the average performance of players born throughout the 1980s. This type of analysis could be used to compare different generations of players or identify peak performance years for a certain cohort.

## Question 4

In [11]:
%%sql
-- Joining People and Batting to get names and stats
-- Summing career runs (R) for each player
SELECT
    p.nameFirst,
    p.nameLast,
    SUM(b.R) AS total_runs
FROM People p
JOIN Batting b ON p.playerID = b.playerID
GROUP BY p.playerID
ORDER BY total_runs DESC
LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('Rickey', 'Henderson', 2295),
 ('Ty', 'Cobb', 2247),
 ('Barry', 'Bonds', 2227),
 ('Babe', 'Ruth', 2174),
 ('Hank', 'Aaron', 2174),
 ('Pete', 'Rose', 2165),
 ('Willie', 'Mays', 2062),
 ('Alex', 'Rodriguez', 2021),
 ('Cap', 'Anson', 1999),
 ('Stan', 'Musial', 1949)]

In [12]:
%%sql
-- Joining People and Batting to get names and stats
-- Summing career hits (H) for each player
SELECT
    p.nameFirst,
    p.nameLast,
    SUM(b.H) AS total_hits
FROM People p
JOIN Batting b ON p.playerID = b.playerID
GROUP BY p.playerID
ORDER BY total_hits DESC
LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('Pete', 'Rose', 4256),
 ('Ty', 'Cobb', 4189),
 ('Hank', 'Aaron', 3771),
 ('Stan', 'Musial', 3630),
 ('Tris', 'Speaker', 3514),
 ('Derek', 'Jeter', 3465),
 ('Cap', 'Anson', 3435),
 ('Honus', 'Wagner', 3420),
 ('Carl', 'Yastrzemski', 3419),
 ('Albert', 'Pujols', 3384)]

4.3 Are there any players who appear in both lists? If so, who are they?
Yes, there are several players on both lists. After comparing the results, the common players are:

Pete Rose

Ty Cobb

Hank Aaron

Stan Musial

Tris Speaker

Derek Jeter

Cap Anson


Analysis: The results showcase the all-time greats of the sport. Creating these leaderboards is a common and powerful use of SQL in sports analytics. The analysis shows a significant overlap between the top run-scorers and top hitters, which is expected since getting a hit is the primary way to get into a position to score a run. Players like Pete Rose, Ty Cobb, and Hank Aaron appearing on both lists confirms their status as elite, well-rounded offensive players. This kind of query is essential for historical comparison and analysis

## Question 5

In [13]:
%%sql
-- Using the provided WITH clause to join CollegePlaying and Batting for Purdue players
WITH joined_data AS (
    SELECT CollegePlaying.playerID, Batting.R
    FROM CollegePlaying
    JOIN Batting ON CollegePlaying.playerID = Batting.playerID
    WHERE CollegePlaying.schoolID = 'purdue'
)
-- Summing the career runs for each Purdue player
SELECT
    playerID,
    SUM(R) AS total_runs
FROM joined_data
GROUP BY playerID
ORDER BY total_runs DESC
LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('allenbe01', 1071),
 ('skowrbi01', 681),
 ('allenje01', 465),
 ('mackife01', 255),
 ('rossyri01', 172),
 ('cianfar01', 136),
 ('tragewa01', 108),
 ('mccabjo01', 6),
 ('kellysp01', 3),
 ('wisehu01', 0)]

In [14]:
%%sql
-- Joining CollegePlaying and Batting
-- Calculating the average runs and grouping by school
SELECT
    cp.schoolID,
    AVG(b.R) as avg_runs
FROM CollegePlaying cp
JOIN Batting b ON cp.playerID = b.playerID
GROUP BY cp.schoolID
ORDER BY avg_runs DESC
LIMIT 10;

 * sqlite:////anvil/projects/tdm/data/lahman/lahman.db
Done.


[('hartford', 86.24528301886792),
 ('txwesleyan', 78.45833333333333),
 ('momaple', 75.1923076923077),
 ('ecentralok', 73.95454545454545),
 ('montevallo', 71.44444444444444),
 ('narizona', 70.15384615384616),
 ('wittnbrgoh', 68.0909090909091),
 ('illchicago', 67.61111111111111),
 ('inconco', 66.32352941176471),
 ('nmmilin', 65.0)]

5.2 Highest number of runs scored by a player from Purdue University
Based on the query above, the highest number of runs scored by a player from Purdue is 1071.


5.4 Which university has the highest average runs scored by its players?
Based on the query results, the university with the highest average runs is SchoolID of Hartford

Analysis: This analysis delves into the amateur origins of professional players. The first query successfully identified the career run totals for the top 10 players from Purdue, with the highest total being 1071. The second query provides a fascinating leaderboard of colleges. It reveals that Hartford has the highest average runs scored by its players who made it to the major leagues. This type of analysis is valuable for scouting and understanding which collegiate programs have historically produced high-performing professional talent.


## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.